Index: src/de/uni_paderborn/fujaba/muml/common/edit/parts/DiscreteMultiPortInstanceBehavior.java
===================================================================
--- src/de/uni_paderborn/fujaba/muml/common/edit/parts/DiscreteMultiPortInstanceBehavior.java	(revision 1426)
+++ src/de/uni_paderborn/fujaba/muml/common/edit/parts/DiscreteMultiPortInstanceBehavior.java	(working copy)
@@ -2,7 +2,14 @@
 
 import org.eclipse.draw2d.IFigure;
 import org.eclipse.draw2d.LayoutListener;
+import org.eclipse.draw2d.LayoutManager;
+import org.eclipse.draw2d.PositionConstants;
+import org.eclipse.gef.editparts.AbstractGraphicalEditPart;
+import org.eclipse.gmf.runtime.diagram.ui.editparts.AbstractBorderItemEditPart;
 import org.eclipse.gmf.runtime.diagram.ui.editparts.GraphicalEditPart;
+import org.eclipse.gmf.runtime.diagram.ui.figures.IBorderItemLocator;
+import org.eclipse.gmf.runtime.diagram.ui.figures.ResizableCompartmentFigure;
+import org.eclipse.gmf.runtime.draw2d.ui.figures.ConstrainedToolbarLayout;
 
 import de.uni_paderborn.fujaba.muml.common.figures.CustomPortFigure;
 import de.uni_paderborn.fujaba.muml.common.figures.CustomPortFigure.PortKind;
@@ -54,37 +61,49 @@
 			 */
 			@Override
 			public void postLayout(IFigure container) {
-//				IBorderItemLocator borderItemLocator = ((AbstractBorderItemEditPart) editPart)
-//						.getBorderItemLocator();
-//				if (borderItemLocator != null) {
-//					borderItemLocator.relocate(editPart.getFigure());
-//					int side = borderItemLocator.getCurrentSideOfParent();
-//					boolean vertical = side == PositionConstants.WEST
-//							|| side == PositionConstants.EAST;
-//					for (Object child : editPart.getChildren()) {
-//						if (child instanceof AbstractGraphicalEditPart) {
-//							IFigure figure = ((AbstractGraphicalEditPart) child)
-//									.getFigure();
-//							if (figure instanceof ResizableCompartmentFigure) {
-//								ResizableCompartmentFigure result = (ResizableCompartmentFigure) figure;
-//
-//								LayoutManager layoutManager = result
-//										.getContentPane().getLayoutManager();
-//								if (layoutManager instanceof ConstrainedToolbarLayout) {
-//									ConstrainedToolbarLayout constrainedToolbarLayout = (ConstrainedToolbarLayout) layoutManager;
-//									boolean oldVerticalValue = !constrainedToolbarLayout.isHorizontal();
-//									if (vertical != oldVerticalValue) {
-//										constrainedToolbarLayout
-//												.setVertical(vertical);
-//										result.validate();
-//									}
-//								}
-//							}
-//						}
-//					}
-//				}
+				// boolean test = false;
+				// if (!test) {
+				// return;
+				// }
+				IBorderItemLocator borderItemLocator = ((AbstractBorderItemEditPart) editPart)
+						.getBorderItemLocator();
+				if (borderItemLocator != null) {
+					borderItemLocator.relocate(editPart.getFigure());
+					int side = borderItemLocator.getCurrentSideOfParent();
+					boolean horizontal = side == PositionConstants.EAST
+							|| side == PositionConstants.WEST;
+					boolean reversed = side == PositionConstants.NORTH
+							|| side == PositionConstants.WEST;
+					for (Object child : editPart.getChildren()) {
+						if (child instanceof AbstractGraphicalEditPart) {
+							IFigure figure = ((AbstractGraphicalEditPart) child)
+									.getFigure();
+							if (figure instanceof ResizableCompartmentFigure) {
+								ResizableCompartmentFigure result = (ResizableCompartmentFigure) figure;
+
+								LayoutManager layoutManager = result
+										.getContentPane().getLayoutManager();
+								if (layoutManager instanceof ConstrainedToolbarLayout) {
+									ConstrainedToolbarLayout constrainedToolbarLayout = (ConstrainedToolbarLayout) layoutManager;
+									if (horizontal != constrainedToolbarLayout
+											.isHorizontal()) {
+										constrainedToolbarLayout
+												.setHorizontal(horizontal);
+
+									}
+									if (reversed != constrainedToolbarLayout
+											.isReversed()) {
+										constrainedToolbarLayout
+												.setReversed(reversed);
+
+									}
+								}
+							}
+						}
+					}
+				}
+				editPart.getFigure().validate();
 			}
 		};
 	}
-
 }
